tests/unit/restapi/v1/test_groups.py

```diff
@@ -0,0 +1,373 @@
+# States Department of Commerce. This software was developed in part by employees of
+# NIST and in part by NIST contractors. Copyright in portions of this software that
+# were developed by NIST contractors has been licensed or assigned to NIST. Pursuant
+# to Title 17 United States Code Section 105, works of NIST employees are not
+# subject to copyright protection in the United States. However, NIST may hold
+# international copyright in software created by its employees and domestic
+# copyright (or licensing rights) in portions of software that were assigned or
+# licensed to NIST. To the extent that NIST holds copyright in this software, it is
+# being made available under the Creative Commons Attribution 4.0 International
+# license (CC BY 4.0). The disclaimers of the CC BY 4.0 license apply to all parts
+# of the software developed or licensed by NIST.
+#
+# ACCESS THE FULL CC BY 4.0 LICENSE HERE:
+# https://creativecommons.org/licenses/by/4.0/legalcode
+"""Test suite for group operations.
+
+This module contains a set of tests that validate the CRUD operations and additional
+functionalities for the group entity. The tests ensure that the groups can be
+registered, queried, and renamed as expected through the REST API.
+"""
+
+from typing import Any
+
+import pytest
+from flask.testing import FlaskClient
+from flask_sqlalchemy import SQLAlchemy
+from werkzeug.test import TestResponse
+
+from dioptra.restapi.routes import V1_GROUPS_ROUTE, V1_ROOT
+
+from ..lib import actions, helpers
+
+# -- Actions ---------------------------------------------------------------------------
+
+
+def modify_group(
+    client: FlaskClient,
+    group_id: int,
+    new_name: str,
+) -> TestResponse:
+    """Rename a group using the API.
+
+    Args:
+        client: The Flask test client.
+        group_id: The id of the group to rename.
+        new_name: The new name to assign to the group.
+        new_description: The new description to assign to the group.
+
+    Returns:
+        The response from the API.
+    """
+    payload = {"name": new_name}
+
+    return client.put(
+        f"/{V1_ROOT}/{V1_GROUPS_ROUTE}/{group_id}",
+        json=payload,
+        follow_redirects=True,
+    )
+
+
+# -- Assertions ------------------------------------------------------------------------
+
+
+def assert_group_response_contents_matches_expectations(
+    response: dict[str, Any], expected_contents: dict[str, Any]
+) -> None:
+    """Assert that group response contents is valid.
+
+    Args:
+        response: The actual response from the API.
+        expected_contents: The expected response from the API.
+
+    Raises:
+        AssertionError: If the response status code is not 200 or if the API response
+            does not match the expected response or if the response contents is not
+            valid.
+    """
+    expected_keys = {
+        "id",
+        "name",
+        "user",
+        "createdOn",
+        "lastModifiedOn",
+    }
```

Comment:
missing the `members` field.

Reply:
Added in commit: [d06c575](https://github.com/usnistgov/dioptra/pull/458/commits/d06c5758eeedc1c5771d2c4ded820b8f4b212ab6)